src/components/collection/Selector/List/index.tsx

```diff
@@ -154,15 +157,25 @@ function CollectionSelectorList({
     }, [filterValue, mappedResourceConfigs]);
 
     useEffect(() => {
-        // Selection disabled
-        if (!selectionEnabled || !setCurrentBinding) {
+        if (
+            // Selection disabled
+            !selectionEnabled ||
+            !setCurrentBinding ||
+            // Filter has not changed so we can skip. Otherwise while the bindings are filtered
+            //  if a user clicks manually on a binding then it will be selected and a split second
+            //  later this effect will run and set it back to a default value.
+            (filterValue.length > 0 &&
+                previousFilterValue &&
+                previousFilterValue.length > 0 &&
+                filterValue === previousFilterValue)
```

Comment:
This fixes the first bug found during testing. Comment explains what it is doing.